src/Algolia.Search/Serializer/AlternativesConverter.cs

```diff
@@ -35,25 +36,33 @@ public class AlternativesConverter : JsonConverter<Alternatives>
         /// <summary>
         /// Read the JSON taking into account the polymorphism of Alternatives
         /// </summary>
-        public override Alternatives ReadJson(JsonReader reader, Type objectType, Alternatives existingValue,
-            bool hasExistingValue, JsonSerializer serializer)
+        public override Alternatives Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            if (reader.TokenType == JsonToken.Null)
-                return null;
+            Alternatives ret;
 
-            if (reader.TokenType == JsonToken.Boolean)
-                return Alternatives.Of(Convert.ToBoolean(reader.Value));
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.Null:
+                    return null;
+                case JsonTokenType.False:
+                case JsonTokenType.True:
+                    ret = Alternatives.Of(reader.GetBoolean());
+                    break;
+                default:
+                    throw new JsonException(
+                        $"Error while reading Token {reader.GetString()} of type {reader.TokenType}.");
```

Comment:
nit: I don't know what is doable/compiles in C# but if you can, I'd recommend to either early return all switch cases or break everywhere and have a single return at the end, but not both. Agree that it's a small example here but it's good practice to ease readability. ~IMHO here, if your compiler permits it, I'd completely remove the temporary `ret` variable and only early return in all cases or throw the exception in the default case.~ After reading the following `switch` statements you have, you seem to most often use the intermediate return value instead of early returns. So for consistency, you may want to use the temporary.